### ISSUE TITLE
Add Rust protections for the private keys

### ISSUE THREAD

**Issue by ipetr0v** [project-oak/oak#4513]
Currently our abstractions have to expose the private keys in order to:
- Pass the private keys from the Restricted Kernel to the enclave application
- Use the `Instance` key to decrypt the `Group` key that was received via Key Provisioning

For example we have the following functions that expose the private key [`instance_encryption_key`](https://github.com/project-oak/oak/blob/28b91c93fe8cfce452b1a9d2177e8ca3adab3b77/oak_containers_orchestrator/src/crypto.rs#L46-L51) and [`get_private_key`](https://github.com/project-oak/oak/blob/28b91c93fe8cfce452b1a9d2177e8ca3adab3b77/oak_crypto/src/encryptor.rs#L59-L61).

We need to implement an abstraction for private keys that:
- Removes the private key from the memory on `drop()`
- Allows encrypting a private key for Key Provisioning
- Also allows sending the private key via an RPC

Though we also don't want to reimplement [Tink](https://developers.google.com/tink).

We also currently have 2 abstractions for the Encryption Private Key:
https://github.com/project-oak/oak/blob/28b91c93fe8cfce452b1a9d2177e8ca3adab3b77/oak_crypto/src/encryptor.rs#L35-L37
https://github.com/project-oak/oak/blob/28b91c93fe8cfce452b1a9d2177e8ca3adab3b77/oak_crypto/src/hpke/mod.rs#L47-L50

We need to merge these implementations into one `EncryptionKey`, and keep it a low level abstraction, since enclave applications don't need to use it.

**Comment by ipetr0v** [comment]
cc @tiziano88 @conradgrobler

**Comment by tiziano88** [comment]
Manually zeroing out memory is actually quite hard, because the compiler will probably optimize it away. We should look for some off the shelf solution to that problem, e.g. https://crates.io/crates/zeroize seems to solve that.

**Comment by conradgrobler** [comment]
Yes, we should replace all instances where we implement `Drop` and use `.fill(0)` to try to zero out keys manually with the zeroize crate.

**Comment by ipetr0v** [comment]
As an addition, we should probably also use [`OPENSSL_cleanse`](https://www.openssl.org/docs/man3.2/man3/OPENSSL_cleanse.html) for our C++ (with Java) code on the client side.
